### PR TITLE
Added JSON schema

### PR DESCRIPTION
**Pull request by sdatspun2**

### CODE REVIEW

**Comment by sdatspun2** [comment]
This PR is for Issue #8

File: schema/json/problem.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      "format": "uri",
+      "description": "A URI reference RFC3986 that identifies the problem type."
+    },
+    "title": {
+      "type": "string",
+      "description": "A short, human-readable summary of the problem type. It SHOULD NOT change from occurrence to occurrence of the problem, except for purposes of localization (e.g., using proactive content negotiation; see RFC7231, Section 3.4)"
+    },
+    "status": {
+      "type": "integer",
+      "description": "The HTTP status code (RFC7231, Section 6) generated by the origin server for this occurrence of the problem."
```
Comment (mnot) [inline schema/json/problem.json:17]: Might be worth adding a `minimum` of `100` and a `maximum` of `599`.

File: schema/json/problem.json [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    },
+    "detail": {
+      "type": "string",
+      "description": "A human-readable explanation specific to this occurrence of the problem."
+    },
+    "instance": {
+      "type": "string",
+      "format": "uri",
```
Comment (mnot) [inline schema/json/problem.json:25]:
```suggestion
      "format": "uri-reference",
```

File: schema/json/problem.json [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "A problem object RFC 7807bis",
+  "type": "object",
+  "properties": {
+    "type": {
+      "type": "string",
+      "format": "uri",
```
Comment (mnot) [inline schema/json/problem.json:8]:
```suggestion
      "format": "uri-reference",
```

File: schema/json/problem.json [hunk 4]
```diff
@@ -0,0 +1,29 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
```
Comment (mnot) [inline schema/json/problem.json:2]: Is this the right URI? I'm not up to speed on the latest in schema-land...

File: schema/json/problem_details.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+          "type": "array",
+          "description": "Problem details if more than one problem exist.",
+          "items": {
+            "$ref": "./problem.json"
+          }
+        }
+      }
+    }
+  ],
+  "x-examples": {
```
Comment (mnot) [inline schema/json/problem_details.json:20]: Shouldn't this be `examples` and an array, as per http://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.9.5 ?

File: schema/json/problem_details.json [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      "properties": {
+        "problems": {
+          "type": "array",
+          "description": "Problem details if more than one problem exist.",
+          "items": {
+            "$ref": "./problem.json"
+          }
+        }
+      }
+    }
```
Comment (mnot) [inline schema/json/problem_details.json:18]: We should remove this until the spec text is in place -- it's not clear we're going to go this way.

File: schema/json/problem_details.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+      "instance": "http://example.com",
+      "problems": [
+        {
+          "type": "string",
+          "title": "string",
+          "status": 0,
+          "detail": "string",
+          "instance": "http://example.com"
+        }
+      ]
```
Comment (mnot) [inline schema/json/problem_details.json:35]: Same as above, should be removed until the issue is settled.

**Review by mnot — CHANGES_REQUESTED** [review]
See inline.

File: schema/json/problem.json (hunk 4 above)
Comment (sdatspun2) [reply schema/json/problem.json:2]: https://tools.ietf.org/html/draft-handrews-json-schema-validation-01#section-5

I kept it slightly old to increase adoption. Latest is 2020-12. Other option could be 2019-09. if not draft7, we could use 2019-09. Lmk.

https://json-schema.org/specification-links.html

File: schema/json/problem_details.json (hunk 5 above)
Comment (sdatspun2) [reply schema/json/problem_details.json:20]: Hmm...relied on a tool. Fixing in next commit.

File: schema/json/problem_details.json (hunk 6 above)
Comment (sdatspun2) [reply schema/json/problem_details.json:18]: Ohk...

File: schema/json/problem.json (hunk 1 above)
Comment (sdatspun2) [reply schema/json/problem.json:17]: Next commit.

File: schema/json/problem.json (hunk 2 above)
Comment (sdatspun2) [reply schema/json/problem.json:25]: Yes. Relative should be supported. Next commit.

File: schema/json/problem.json (hunk 3 above)
Comment (sdatspun2) [reply schema/json/problem.json:8]: +1

File: schema/json/problem.json (hunk 4 above)
Comment (dret) [reply schema/json/problem.json:2]: not sure about the "best" version/URI to choose, as they're still in eternal draft mode.

File: schema/json/problem_details.json (hunk 6 above)
Comment (dret) [reply schema/json/problem_details.json:18]: "going to go this way" meaning adding a JSON schema, @mnot?

Comment (mnot) [reply schema/json/problem_details.json:18]: No, I mean we haven't resolved how multiple problems should be handled.

Comment (sdatspun2) [reply schema/json/problem_details.json:18]: The schema proposal is an attempt to resolve it :)

File: schema/json/problem.json (hunk 4 above)
Comment (sdatspun2) [reply schema/json/problem.json:2]: I am thinking of removing it. Some linters do not recognize $schema, esp. those that follow pre OpenAPI 3.1 spec.

Comment (darrelmiller) [reply schema/json/problem.json:2]: We use the following 
```
 "$schema": "https://json-schema.org/draft/2020-12/schema",
```
in the [OpenAPI 3.1 metaschema](https://github.com/OAI/OpenAPI-Specification/blob/main/schemas/v3.1/schema.json) and it has been reviewed by the JSON Schema editors.

File: schema/json/problem_details.json (hunk 6 above)
Comment (mnot) [reply schema/json/problem_details.json:18]: Right, but this is a proposal to resolve the issue about having a JSON Schema, not that. Addressing two issues together conflates them, and we haven't yet come to consensus on the latter. 

In this situation, the usual process is to address the more straightforward thing first -- in this case, adding a JSON Schema -- and then adding to it when we resolve the other issue. So I'd suggest removing `properties` for now (although we might add it back later).

File: schema/json/problem.json (hunk 4 above)
Comment (mnot) [reply schema/json/problem.json:2]: Sounds good to me - let's change to that value.

File: schema/json/problem_details.json (hunk 6 above)
Comment (sdatspun2) [reply schema/json/problem_details.json]: Removed

File: schema/json/problem_details.json (hunk 7 above)
Comment (sdatspun2) [reply schema/json/problem_details.json]: Removed

File: schema/json/problem_details.json (hunk 5 above)
Comment (sdatspun2) [reply schema/json/problem_details.json]: not relevant. removed.

File: schema/json/problem.json (hunk 4 above)
Comment (sdatspun2) [reply schema/json/problem.json]: Changed

**Review by mnot — APPROVED** [review]
LGTM

**Pull request merged**